test/e2e/upgrades/ingress.go

```diff
@@ -173,6 +181,8 @@ func (t *IngressUpgradeTest) verify(f *framework.Framework, done <-chan struct{}
 	// WaitForIngress() tests that all paths are pinged, which is how we know
 	// everything is synced with the cloud.
 	t.jig.WaitForIngress(false)
+	// Stop performing the continuous ping we started earlier since the updated controller is fully synced.
+	continuousPing <- struct{}{}
```

Comment:
According to the `wait.Until` description, it expects the channel to be closed.
https://godoc.org/k8s.io/apimachinery/pkg/util/wait#WaitFunc

Reply:
Fixed.